### PR TITLE
JSON.parse error fix on nebulas network error

### PR DESCRIPTION
**Pull request by headersalreadysent**
When mainnet or testnet gave error or became unreachable cloudflare returns html error pages for 502, 522 http error codes. 

ResponseTransform of axios cant handle that and only show unable to parse json errror. Instead of this code can show which request get error and what kind of error happened.

I change httprequest.js and add some control over axios thenable.

### CODE REVIEW

File: lib/httprequest.js [hunk 1]
```diff
@@ -5,59 +5,66 @@ var axios = require("axios");
 var debugLog = false;
 
 var HttpRequest = function (host, timeout, apiVersion) {
-    this.host = host || "http://localhost:8685";
-    this.timeout = timeout || 0;
-    this.apiVersion = apiVersion || "v1";
+	this.host = host || "http://localhost:8685";
+	this.timeout = timeout || 0;
+	this.apiVersion = apiVersion || "v1";
```
Comment (yupnano) [inline lib/httprequest.js]: should avoid unnecessary changes, such as indent style.

Comment (headersalreadysent) [reply lib/httprequest.js]: Sorry visual studio code set as format on save. İ can fix.
